src/components/PeopleFilters.tsx

```diff
@@ -1,18 +1,56 @@
+import { ChangeEvent, useState } from 'react';
+import { SearchLink } from './SearchLink';
+import { getSearchWith } from '../utils/searchHelper';
+import { useSearchParams } from 'react-router-dom';
+import classNames from 'classnames';
+
+const ageParams = ['16', '17', '18', '19', '20'];
+
 export const PeopleFilters = () => {
+  const [query, setQuery] = useState('');
+  const [searchParams, setSearchParams] = useSearchParams();
+  const ages = searchParams.getAll('centuries');
+  const sex = searchParams.get('sex') || null;
+
+  const handleQueryChange = (e: ChangeEvent<HTMLInputElement>) => {
+    const serch = getSearchWith(searchParams, { query: e.target.value });
+
+    setSearchParams(serch);
+    setQuery(e.target.value);
+  };
+
+  const addAge = (age: string) => {
+    return ages.includes(age)
+      ? ages.filter(currAge => currAge !== age)
+      : [...ages, age];
+  };
+
+  const toggleAges = (age: string) => {
+    const newAges = addAge(age);
+
+    setSearchParams({ centuries: newAges });
```

Comment:
The `setSearchParams` call in `toggleAges` should preserve existing search parameters. Currently, it only sets the `centuries` parameter, which might remove other parameters like `sex` or `query`. Consider using `getSearchWith` to update the `centuries` parameter while preserving others.